src/trousse/dataset.py

```diff
@@ -487,6 +488,79 @@ def operations_history(self) -> OperationsList:
         """
         return self._operations_history
 
+    @property
+    def to_be_fixed_cols(self) -> Set[str]:
+        """
+        Return name of the columns containing values of mixed types.
+
+        Returns
+        -------
+        Set[str]
+            Set of columns with values of different types
+        """
+        return self._columns_type.mixed_type_cols
+
+    @property
+    def to_be_encoded_cat_cols(self):
+        """
+        Find categorical columns that needs encoding.
+
+        It also checks if they are already encoded.
+
+        Returns
+        -------
+        Set[str]
+            Set of categorical column names that need encoding
+
+        """
+        to_be_encoded_categorical_cols = set()
+        # TODO: Check this because maybe categorical columns that are numerical, do
+        #  not need encoding probably!
+        categorical_cols = self.str_categorical_cols | self.num_categorical_cols
+        for categ_col in categorical_cols:
+            if self.get_enc_column_from_original(categ_col) is None:
+                to_be_encoded_categorical_cols.add(categ_col)
+
+        return to_be_encoded_categorical_cols
+
+    # =====================
+    # =    METHODS        =
+    # =====================
+    def _data_to_mixed_types(self, df: pd.DataFrame):
+        """
+        Transform 'object'-typed column values to appropriate types.
+
+        This static method analyzes the DataFrame ``df`` columns that have
+        dtype='object'. The columns that have numeric and string values are interpreted
+        and casted by Pandas as columns with dtype='object' and all the numeric,
+        boolean or datetime values are transformed into string typed values
+        (e.g. 2 -> '2').
+        This would disturb the Dataset column type inference, so this method restore
+        numeric, boolean and datetime values to the appropriate types for columns
+        with dtype='object'.
+
+        Parameters
+        ----------
+        df : pd.DataFrame
+            DataFrame containing the columns that will be analyzed. It will not be
+            modified inplace.
+
+        Returns
+        -------
+        pd.DataFrame
+            New DataFrame with numeric and boolean values restored to the appropriate
+            types for columns with dtype='object'.
+        """
+        mixed_df = df.copy()
+        str_cols = df.select_dtypes(include="object").columns
+        for col in str_cols:
+            mixedtype_converter = _DfConvertToMixedType(column=col)
```

Comment:
`_DfConvertToMixedType ` is **private** and we cannot use it in the dataset module. If we need to use that here let's rename it